### PR TITLE
Rewrite getCookies to be more specific and allow getting a cookie by nam...

### PR DESCRIPTION
**Pull request by AutomatedTester**
...e

The change allows us to getCookies and getCookies(name) and return them to the remote end. The notable change is the change from
GET to POST to allow a body that can be parsed to get the cookie name

### CODE REVIEW

File: 01_introduction.html [hunk 1]
```diff
@@ -113,5 +113,8 @@
     and <code><a href=http://www.ecma-international.org/ecma-262/5.1/#sec-15.1.2.3>parseFloat</a></code>
     from [[!ECMA-262]] with the exception that non-valid integer
     or float return values are returned as 0 or 0.0.
+
+    <h4>Top Level Browsing Context no longer open</h4>
+    <p>A top level browsing context is <dfn>no longer open</dfn> if it has been <a href="https://html.spec.whatwg.org/#a-browsing-context-is-discarded">discarded</a>
```
Comment (andreastt) [inline 01_introduction.html]: Can't we just use the definition from HTML, that a browser context has been “discarded”?

File: 12_cookies.html [hunk 2]
```diff
@@ -38,20 +20,56 @@
...
-        </ul>
-        <dl class='parameters'>
-          <dt>optional DOMString? name</dt>
-          <dd>The name (e.g. <code>name=</code>) of the cookie that needs to be returned.</dd>
-        </dl>
+          <p>The <a>remote end steps</a> for the <a>getCookie</a> command are:</p>
+          <ol>
+            <li><p>If the <a>current browsing context</a> is <a>no longer open</a>, return an error with code <a>no such window</a>.</p></li>
+            <li><p>Let <var>name</var> be the result of <a>getting a property</a> named "name" from the <var>parameters</var> argument.</p></li>
+            <li><p>Let <var>result</var> be an empty JSON List</p></li>
```
Comment (andreastt) [inline 12_cookies.html]: s/List/list/ and missing full period.

File: 12_cookies.html [hunk 3]
```diff
@@ -38,20 +20,56 @@
...
+            <li><p>Let <var>cookies</var> be a list of all cookies [[!RFC6265]] in the cookie store associated with <a href="https://dom.spec.whatwg.org/#concept-document-url"> the current document or frame address</a></p></li>
```
Comment (andreastt) [inline 12_cookies.html]: Missing full stop.

File: 12_cookies.html [hunk 4]
```diff
@@ -38,20 +20,56 @@
...
+            <li><p>Let <var>length</var> be the length of <var>cookies</var>.</p></li>
+            <li><p>Let <var>k</var> be 0</p></li>
+            <li><p>While <var>k</var> < <var>length</var>: </p>
```
Comment (andreastt) [inline 12_cookies.html]: s/</&lt;/

File: 12_cookies.html [hunk 5]
```diff
@@ -38,20 +20,56 @@
...
+              <ol>
+                <li><p>Let <var>cookie</var> be the value in <var>cookies</var> at index <var>k</var>.</p></li>
+                <li><p>If <var>name</var> is undefined:</p>
+                  <ol>
+                    <li><p>Append a <a>serialised cookie</a> to <var>result</var>.</p></li>
```
Comment (andreastt) [inline 12_cookies.html]: s/serialised/serialized/

File: 12_cookies.html [hunk 6]
```diff
@@ -38,20 +20,56 @@
...
+                  </ol>
+                </li>
+                <li><p>if <var>name</var> is defined and is equal to <var>cookie-name</var>:</p>
```
Comment (andreastt) [inline 12_cookies.html]: s/if/If/

File: 12_cookies.html [hunk 7]
```diff
@@ -38,20 +20,56 @@
...
+                  <ol>
+                    <li><p>Append a <a>serialised cookie</a> to <var>result</var>.</p></li>
+                  </ol>
+                </li>
+                <li><p>Increase the value of <var>k</var> by 1</p></li>
```
Comment (andreastt) [inline 12_cookies.html]: This doesn't jump back to continue the loop, but just increases `k`.  I strongly suggest we just say “iterate through the list” and “for each `<var>cookie</var>`, do:”.

File: 12_cookies.html [hunk 8]
```diff
@@ -38,20 +20,56 @@
...
+              </ol>
+            </li>
+            <li><p>Return <a>success</a> with data <var>result</var></p></li>
```
Comment (andreastt) [inline 12_cookies.html]: Missing full stop.

File: 12_cookies.html [hunk 9]
```diff
@@ -38,20 +20,56 @@
...
+          </ol>
+          <p>A <dfn>serialised cookie</dfn> is created with the following algorithm:</p>
+          <p>
+            <ol>
+              <li><p>Let <var>serialised cookie</var> be an empty map.</p></li>
```
Comment (andreastt) [inline 12_cookies.html]: a map left initially empty

File: 12_cookies.html (hunk 7 above)
Comment (AutomatedTester) [inline 12_cookies.html]: This follows the same format for `forEach` so will be keeping this as is. Adding `Go To Step x` is frowned upon, apparently, unless it is for highly complex loops.

File: 12_cookies.html (hunk 6 above)
Comment (AutomatedTester) [inline 12_cookies.html]: in [rfc6265](http://tools.ietf.org/html/rfc6265#section-4.1.1) in the `cookie-pair` is `cookie-name = cookie-value` so would like to preserve some of the semantics by using `cookie-name`

**Comment by AutomatedTester** [comment]
landed in 08382499cb775645b61b313be4fd441b72259b56

**Pull request closed without merging**